### PR TITLE
DRIVERS-2715 Investigate Increased Costs of Drivers Atlas QA Usage

### PR DESCRIPTION
**Pull request by blink1073**
The clusters were silently failing to delete because they were trying to delete the cluster in `drivers-atlas-testing` instead of the the newly created project.

### CODE REVIEW

**Comment by blink1073** [comment]
It looks like these failures were pre-existing in https://github.com/mongodb-labs/drivers-atlas-testing/pull/161.

File: astrolabe/cli.py [hunk 1]
```diff
@@ -613,8 +615,11 @@ def delete_test_cluster(ctx, spec_test_file, workload_file, org_id, project_name
     if project:
         try:
             ctx.obj.client.groups[project.id].clusters[cluster_name].delete()
-        except AtlasApiBaseError:
-            pass
+            print(f"{msg} done.")
+        except AtlasApiBaseError as e:
+            pprint(e)
```
Comment (matthewdale) [inline astrolabe/cli.py:620]: Should we re-raise this so the task fails if deleting the cluster fails? Is there something else we can do to make it more obvious that we've left an expensive resource hanging around?

File: astrolabe/cli.py [hunk 2]
```diff
@@ -613,8 +615,11 @@ def delete_test_cluster(ctx, spec_test_file, workload_file, org_id, project_name
...
+    else:
+        print(f"Project {project_name} not found!")
```
Comment (matthewdale) [inline astrolabe/cli.py:622]: Should we raise an exception here if we attempted to delete a cluster but couldn't find the project?

File: astrolabe/cli.py (hunk 1 above)
Comment (blink1073) [reply astrolabe/cli.py:620]: This occurs during a tear-down tastk so failing wouldn't fail the build.

File: astrolabe/cli.py (hunk 2 above)
Comment (blink1073) [reply astrolabe/cli.py:622]: Same answer as above

File: astrolabe/cli.py (hunk 1 above)
Comment (matthewdale) [reply astrolabe/cli.py:620]: Good point. We could use `post_error_fails_task: true` to make post functions fail the task.

**Review by matthewdale — APPROVED** [review]
Nice fix! 👍

File: astrolabe/cli.py (hunk 1 above)
Comment (matthewdale) [reply astrolabe/cli.py:620]: Created [DRIVERS-2729](https://jira.mongodb.org/browse/DRIVERS-2729) to track that suggestion.

**Pull request merged**